subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/TasksFromProjectDependencies.java

```diff
@@ -27,16 +28,16 @@
 
 class TasksFromProjectDependencies extends AbstractTaskDependency {
     private final String taskName;
-    private final DependencySet dependencies;
+    private final Configuration configuration;
 
-    public TasksFromProjectDependencies(String taskName, DependencySet dependencies) {
+    public TasksFromProjectDependencies(String taskName, Configuration configuration) {
```

Comment:
Ah, I didn't consider `Supplier`.  I'll try that.

I wanted to fix this so that any calls to getTaskDependencyFromProjectDependency would behave in a similar way.